tests/ssl-opt.sh

```diff
@@ -9197,11 +9197,14 @@ requires_config_enabled MBEDTLS_DEBUG_C
 requires_config_enabled MBEDTLS_SSL_CLI_C
 run_test    "TLS 1.3: CertificateRequest check - openssl" \
```

Comment:
This test has done its job while we supported only the CertificateRequest message. Now it is a duplicate of a test below, thus we can just remove it.